internal/engine/upper.go

```diff
@@ -49,17 +92,23 @@ func (u Upper) Up(ctx context.Context, service model.Service, watchMounts bool,
 		}
 
 		for {
-			// TODO(matt) buffer events a bit so that we're not triggering 10 builds when you change branches
 			select {
 			case err := <-watcher.Errors():
 				return err
 			case event := <-watcher.Events():
-				logger.Get(ctx).Info("file changed, rebuilding %v", service.Name)
-				path, err := filepath.Abs(event.Name)
-				if err != nil {
-					return err
+				logger.Get(ctx).Info("files changed, rebuilding %v", service.Name)
+				events := []fsnotify.Event{event}
+				events = append(events, u.coalesceEvents(watcher.Events())...)
```

Comment:
This feels weird to me. Instead of appending in this receive case, would it make more sense to make a new channel of coalesced events that this part of the code receives from?

Reply:
yeah, that's prettier